xdsl/dialects/ltl.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from typing import Annotated
+
+from xdsl.dialects.builtin import IntegerType, Signedness
+from xdsl.ir import Attribute, Dialect, ParametrizedAttribute, SSAValue, TypeAttribute
+from xdsl.irdl import (
+    AnyOf,
+    ConstraintVar,
+    IRDLOperation,
+    irdl_attr_definition,
+    irdl_op_definition,
+    result_def,
+    var_operand_def,
+)
+
+"""
+Implementation of the LTL dialect by CIRCT. Documentation: https://circt.llvm.org/docs/Dialects/LTL/
+"""
+
+
+@irdl_attr_definition
+class Property(ParametrizedAttribute, TypeAttribute):
+    """
+    The `ltl.property` type represents a verifiable property built from linear
+    temporal logic sequences and quantifiers, for example, *"if you see sequence
+    A, eventually you will see sequence B"*.
+    Note that this type explicitly identifies a *property*. However, a boolean
+    value (`i1`) or a sequence (`ltl.sequence`) is also a valid property.
+    Operations that accept a property as an operand will use the `AnyProperty`
```

Comment:
I guess `AnyProperty` is no longer a thing? I would keep implementation details out of docstrings, they are usually not as helpful as describing what happens in more concise terms.